src/v/cluster/persisted_stm.cc

```diff
@@ -168,18 +179,31 @@ ss::future<bool> persisted_stm::sync(model::timeout_clock::duration timeout) {
           deadline, [] { return false; });
     }
     _is_catching_up = true;
+
+    co_await _c->refresh_commit_index();
+
     auto term = _c->term();
     auto is_synced = false;
     try {
-        co_await quorum_write_empty_batch(model::timeout_clock::now() + timeout)
-          .discard_result();
-        is_synced = term == _c->term();
```

Comment:
it looks like the commit that switches to co-routines should be updated to _only_ switch to coroutines, because here the apparent change in semantics is all re-written.